### PR TITLE
Feature/more notification templaters

### PR DESCRIPTION
**Pull request by Lex-23**

### CODE REVIEW

**Comment by Lex-23** [comment]
[Upgrading notification system](https://trello.com/c/rI1ECV4R/32-upgrading-notification-system)

File: cryptostock_api/cryptostock/notification/models.py [hunk 1]
```diff
@@ -90,38 +45,94 @@ def send(self, message):
...
 
     @classmethod
-    def get(cls, consumer_type, notification_type):
-        # TODO: get templator logic
-        return BaseTemplator
+    def get(cls, notification_type, consumer_type):
+        return cls._templaters.get(
+            (notification_type, consumer_type),
+            cls._templaters.get((notification_type, None), BaseTemplater),
+        )
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/notification/models.py:84]:
```suggestion
        return (
            cls._templaters.get((notification_type, consumer_type))
            of cls._templaters.get((notification_type, None))
            or BaseTemplater
        )
```

File: cryptostock_api/cryptostock/utils/notification_handlers/email_client.py [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+
+def email_notify(message, context):
+    # TODO: validate context
+    if type(message) == dict:
+        recipient = context.get("recipient", message["recipient"])
+        subject = message["subject"]
+        message = message["body"]
+    else:
+        recipient = context.get("recipient")
+        subject = f"Notification from {CRYPTOSTOCK_NAME}"
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/utils/notification_handlers/email_client.py:17]: When message is not dict?

Comment (Lex-23) [reply cryptostock_api/cryptostock/utils/notification_handlers/email_client.py:17]: When we have only notification_type for templater

File: cryptostock_api/cryptostock/utils/notification_handlers/email_client.py [hunk 3]
```diff
@@ -1,6 +1,26 @@
...
 from django.conf.global_settings import EMAIL_HOST_USER
 from django.core.mail import send_mail
 
+CRYPTOSTOCK_NAME = os.environ["CRYPTOSTOCK_NAME"]
+
+
+def email_notify(message, context):
+    # TODO: validate context
+    if type(message) == dict:
+        recipient = context.get("recipient", message["recipient"])
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/utils/notification_handlers/email_client.py:12]: I think `recipient` should be always in context

File: cryptostock_api/cryptostock/notification/models.py (hunk 1 above)
Comment (Lex-23) [reply cryptostock_api/cryptostock/notification/models.py:84]: got it

File: cryptostock_api/cryptostock/utils/notification_handlers/email_client.py (hunk 3 above)
Comment (Lex-23) [reply cryptostock_api/cryptostock/utils/notification_handlers/email_client.py:12]: agree

File: cryptostock_api/cryptostock/utils/notification_handlers/email_client.py [hunk 4]
```diff
@@ -1,6 +1,25 @@
...
+
+
+def email_notify(message, context):
+    # TODO: validate context
+    recipient = context.get("recipient")
+    if type(message) == dict:
+        subject = message["subject"]
+        message = message["body"]
+    else:
+        subject = f"Notification from {CRYPTOSTOCK_NAME}"
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/utils/notification_handlers/email_client.py:16]:
```suggestion
    subject = f"Notification from {CRYPTOSTOCK_NAME}"
    recipient = context.get("recipient")
    if isinstance(message, dict):
        subject = message["subject"]
        message = message["body"]
```

**Review by vadim-denisov — APPROVED** [review]

**Pull request merged**